Game/source/controllers/MouseController.hx

```diff
@@ -45,22 +48,27 @@ class MouseController {
         //if mouse is released and there's no object there
         if (FlxG.mouse.justReleased) {
             if (selectedSprite == null && !wasJustReleased) {
-                var turret:TowerController = new TowerController(FlxG.mouse.x, FlxG.mouse.y, 40, 150, 400);
+                var turret:Tower = state.towerController.buildTower(towerPreset, ammoType, FlxG.mouse.x, FlxG.mouse.y);
                 turret.updateHitbox();
-                state.add(turret);
+                GameState.towers.push(turret);
                 spriteList.push(turret);
                 wasJustReleased = true; 
             }
             selectedSprite = null; 
         }
     }
 
-    public function setState(state:FlxState):Void { 
+    public function setState(state:GameState):Void { 
         this.state = state; 
     }
 
-    public function new(state:FlxState):Void {
+    public function new(state:GameState):Void {
         this.state = state; 
+
+        towerPreset.add(new GunBase(100, 400, "normal", 1, 1));
```

Comment:
Feel free to throw out any changes I made to MouseController. This was just so I could test what I wrote.